src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java

```diff
@@ -0,0 +1,248 @@
+package org.broadinstitute.hellbender.tools;
+
+import htsjdk.tribble.Tribble;
+import htsjdk.tribble.index.Index;
+import htsjdk.tribble.index.IndexFactory;
+import htsjdk.tribble.index.linear.LinearIndex;
+import htsjdk.tribble.index.tabix.TabixIndex;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+public final class IndexFeatureFileIntegrationTest extends CommandLineProgramTest {
+
+    @Test
+    public void testVCFIndex() {
+        final File ORIG_FILE = new File(getTestDataDir(), "test_variants_for_index.vcf");
+        final File outName = BaseTest.createTempFile("test_variants_for_index.", ".idx");
+
+        final String[] args = new String[]{
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+                "-O" ,  outName.getAbsolutePath()
+        };
+        final Object res = this.runCommandLine(args);
+        Assert.assertEquals(res, outName.getAbsolutePath());
+
+        final Index index = IndexFactory.loadIndex(res.toString());
+        Assert.assertTrue(index instanceof LinearIndex);
```

Comment:
The type of an index on a regular (non-GVCF) vcf could actually be either `LinearIndex` or `IntervalTreeIndex`, as `IndexFactory.createDynamicIndex()` with `IndexFactory.IndexBalanceApproach.FOR_SEEK_TIME` will (I believe) create both a `LinearIndexCreator` and an `IntervalIndexCreator`, score the resulting indices, and choose the best one.

Not sure how to reflect this in tests though, as I don't know what input would trigger the selection of a `IntervalTreeIndex` over a `LinearIndex`.